spinalcordtoolbox/qmri/mt.py

```diff
@@ -101,8 +101,8 @@ def compute_mtsat(nii_mt, nii_pd, nii_t1,
     if nii_t1map is None:
         # compute R1
         logger.info("Compute T1 map...")
-        r1map = 0.5 * np.true_divide((fa_t1_rad / tr_t1) * nii_t1.data - (fa_pd_rad / tr_pd) * nii_pd.data,
-                                     nii_pd.data / fa_pd_rad - nii_t1.data / fa_t1_rad)
+        r1map = 0.5 * np.multiply(np.multiply(nii_b1map.data, nii_b1map.data), np.true_divide((fa_t1_rad / tr_t1) * nii_t1.data - (fa_pd_rad / tr_pd) * nii_pd.data,
+                                     nii_pd.data / fa_pd_rad - nii_t1.data / fa_t1_rad))
```

Comment:
Thank you for the fix!! Admittedly, I don't know if I'm the best person to review the substance of the PR, but I'm happy to take care of the failing CI tests.

Reply:
It seems that currently `nii_b1map` is an optional argument to our CLI script:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/27e659857333a77790ea861d979a26569ae4b7d8/spinalcordtoolbox/scripts/sct_compute_mtsat.py#L168-L171

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/27e659857333a77790ea861d979a26569ae4b7d8/spinalcordtoolbox/qmri/mt.py#L62-L65

Because it is optional, there are cases where the B1 map won't be provided; the current PR fails in such cases.

So, I've refactored the PR such that the old computation will be preserved, and the new B1 correction will be applied on a separate line (i.e. only if `nii_b1map` is present).

Reply:
>>Assuming that my small refactor produces the results you expect (could you double-check on your mouse data?)

Done - results reproduce exactly! Thanks for the test fix!